lms/djangoapps/grades/models.py

```diff
@@ -652,6 +651,10 @@ class Meta(object):
     possible_all_override = models.FloatField(null=True, blank=True)
     earned_graded_override = models.FloatField(null=True, blank=True)
     possible_graded_override = models.FloatField(null=True, blank=True)
+    # store the source of the system that caused the override
+    system = models.CharField(max_length=100, blank=True, null=True)
+    # store the reason for the override
+    override_reason = models.CharField(max_length=300, blank=True, null=True)
```

Comment:
nit: maybe consider letting this max_length be longer (1000, maybe) - but if there's already been some product input on this length, then ignore my comment.